homeassistant/components/tplink/light.py

```diff
@@ -252,19 +238,19 @@ async def async_turn_off(self, **kwargs: Any) -> None:
         await self.device.turn_off(transition=transition)
 
     @property
-    def min_mireds(self) -> int:
+    def min_color_temp_kelvin(self) -> int:
         """Return minimum supported color temperature."""
-        return kelvin_to_mired(self.device.valid_temperature_range.max)
+        return cast(int, self.device.valid_temperature_range.min)
 
     @property
-    def max_mireds(self) -> int:
+    def max_color_temp_kelvin(self) -> int:
         """Return maximum supported color temperature."""
-        return kelvin_to_mired(self.device.valid_temperature_range.min)
+        return cast(int, self.device.valid_temperature_range.max)
 
     @property
-    def color_temp(self) -> int | None:
-        """Return the color temperature of this light in mireds for HA."""
-        return kelvin_to_mired(self.device.color_temp)
+    def color_temp_kelvin(self) -> int:
+        """Return the color temperature of this light."""
+        return cast(int, self.device.color_temp)
```

Comment:
Unsure why casting here is required (mypy thinks it's `Any`) as the type is defined here: https://github.com/python-kasa/python-kasa/blob/master/kasa/smartbulb.py#L390

Potentially because of the `type: ignore` used for the property decorator?

Reply:
I think `requires_update` is the decorator causing issues for `mypy`, as that is not typed.

Reply:
I'm not sure how the decorator should be generalized to make it use the types from underlying properties, so I suppose it's fine to use cast here for the time being.

Reply:
Thanks for the tip @epenet, created https://github.com/python-kasa/python-kasa/issues/435 :)